realm-annotations-processor/src/main/java/io/realm/processor/Utils.java

```diff
@@ -73,7 +76,52 @@ public static boolean isString(VariableElement field) {
     }
 
     /**
-     * Returns true if a given field type string is "java.lang.String", false otherwise.
+     * Returns {@code true} if a field is a primitive type, {@code false} otherwise.
+     */
+    public static boolean isPrimitiveType(String typeString) {
+        if (typeString == null) {
```

Comment:
`typeString` should never be null, we should not continue the annotation processor